pgjdbc/src/test/java/org/postgresql/test/jdbc2/RefCursorTest.java

```diff
@@ -172,4 +176,37 @@ public void testResultType() throws SQLException {
     call.close();
   }
 
+  private void checkAndRunResult(Integer fetchSize) throws SQLException {
+    assumeCallableStatementsSupported();
+    CallableStatement call = con.prepareCall("{ ? = call testspg__getRefcursor () }");
+    call.registerOutParameter(1, cursorType);
+    if (fetchSize != null) {
+      call.setFetchSize(fetchSize);
+    }
```

Comment:
Frankly speaking, I would not expect `call.setFetchSize` to be inherited when fetching the refcursor.

Reply:
I'm not sure why not ?

Statement.setFetchSize() docs say 
```
Gives the JDBC driver a hint as to the number of rows that should be fetched from the database when more rows are needed for ResultSet objects generated by this Statement. If the value specified is zero, then the hint is ignored. The default value is zero.
Params:
rows – the number of rows to fetch
Throws:
SQLException – if a database access error occurs, this method is called on a closed Statement or the condition rows >= 0 is not satisfied.
```

How else would they be used ?

Reply:
Yes, however it is allowed. 
```
Gives the JDBC driver a hint as to the number of rows that should be fetched from the database when more rows are needed for this ResultSet object. If the fetch size specified is zero, the JDBC driver ignores the value and is free to make its own best guess as to what the fetch size should be. The default value is set by the Statement object that created the result set. The fetch size may be changed at any time.
```

I don't see a way to change it once we have the statement. Does that currently work with portals ?